src/JSInterop/Microsoft.JSInterop/src/Infrastructure/DotNetDispatcher.cs

```diff
@@ -292,6 +295,32 @@ public static void EndInvokeJS(JSRuntime jsRuntime, string arguments)
             }
         }
 
+        /// <summary>
+        /// Accepts the byte array data being transferred from JS to DotNet.
+        /// </summary>
+        /// <param name="jsRuntime">The <see cref="JSRuntime"/>.</param>
+        /// <param name="id">Identifier for the byte array being transfered.</param>
+        /// <param name="data">Byte array to be transfered from JS.</param>
+        public static void SupplyByteArray(JSRuntime jsRuntime, long id, byte[] data)
+        {
+            if (id == 0)
+            {
+                // Starting a new transfer, clear out previously stored byte arrays
+                jsRuntime.ResetByteArraysToBeRevived();
+            }
```

Comment:
I think we can probably clear the buffers earlier, like after the `BeginInvokeDotNet` call has deserialized the arguments. This is important because the list will be released right after we are done using them and we will avoid holding on to them for longer than necessary.

Reply:
I think it does also do that already. The call here is only to guarantee we clean up if something went wrong during the last call.

Reply:
Wouldn't doing it on a `finally` block the CircuitHost achieve the same result? If something went wrong with the previous call, we should fail and tear-down the circuit, shouldn't we?

Reply:
If we'd like to do it from the circuit host, it'd require us to add `ResetByteArraysToBeRevived` to the public API. Would that be acceptable?

Reply:
My personal preference would be not making it part of the public API. Given the discussion at https://github.com/dotnet/aspnetcore/pull/33015#discussion_r639619461 it seems like we're OK keeping the cleanup as-is.